cubic-server/command-parser/commands/help.hpp

```diff
@@ -0,0 +1,100 @@
+#ifndef CUBICSERVER_COMMANDPARSER_COMMANDS_HELP_HPP
+#define CUBICSERVER_COMMANDPARSER_COMMANDS_HELP_HPP
+
+#include "logging/Logger.hpp"
+
+#include <vector>
+#include <unordered_map>
+#include <functional>
+#include <string>
+
+namespace help {
+
+    void help(std::vector<std::string>& args);
+    void execute(std::vector<std::string>& args);
+
+    //TODO: Move this code
+    static const std::unordered_map<std::string, std::function<void(std::vector<std::string>&)>> helps {
+            {"help", help::help}
+    };
+
+    static const std::vector<std::string> allCommands = {
+        "/advancement (grant|revoke)",
```

Comment:
Can plugins add commands ?
If so, we need a way to add them to the help command too. It would be better to have a command class that can be added to a registry or something, event if they just contain static function.

Reply:
Do you mean something like that ? 
```cpp
namespace help_test {

struct helptest
{
    static void help(std::vector<std::string>& args);
    static void execute(std::vector<std::string>& args);
    inline static const std::string name = "/help [<command>]";
};

static const std::unordered_map<std::string, std::function<void(std::vector<std::string>&)>> helps {
    {"help", help_test::helptest::help},
    {"?", help_test::helptest::help}
};

static const std::vector<std::string> allCommands = {
    help_test::helptest::name,
};
}
```

Reply:
I still need to work on a more reliable way of storing these map/vector 'cause each function will have to be included in the help.hpp to be in these map/vector

Reply:
Not really. Basically, we need a base command class with at least 3 virtual commands : `execute`, `help` and `autocomplete`.

For now, we just need `execute` and `help`. The will do the same as what you did, but the advantage, is that we can store commands, add some (via plugins for example), override some, etc...

This is useful in your case, because you can see what command is stored, and create the help message with only those.